packages/utils/src/time.ts

```diff
@@ -135,14 +135,27 @@ export const browserPerformanceTimeOrigin = ((): number | undefined => {
   if (!performance) {
     return undefined;
   }
-  if (performance.timeOrigin) {
+  // If performance APIs are over an hour skewed don't use them
+  const THRESHOLD = 3600 * 1000;
+  // eslint-disable-next-line deprecation/deprecation
+  const navigationStart = performance.timing && performance.timing.navigationStart;
+  const hasNavigationStart = typeof navigationStart === 'number';
+  // Unfortunately browsers may report an inaccurate time origin data, which results in poor results in performance
+  // data. We treat time origin data as reliable if they are either within a reasonable threshold of the current time,
+
+  const timeOriginIsReliable = Math.abs(performance.timeOrigin + performance.now() - Date.now()) < THRESHOLD;
```

Comment:
I think without checking for undefined `performance.timeOrigin` this would break in for example Safari, right?

```suggestion
  const timeOriginIsReliable = performance.timeOrigin && Math.abs(performance.timeOrigin + performance.now() - Date.now()) < THRESHOLD;
```

Reply:
Yeah if `performance.timeOrigin` is undefined this will evaluate to `NaN < THRESHOLD` => `false`